### PR TITLE
Add code to calculate COS S_REGION and put into keyword

### PR DESCRIPTION
**Pull request by stscirij**

### CODE REVIEW

**Review by raswaters — APPROVED** [review]

File: costools/add_cos_s_region.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        try:
+            ra_aper = ext1hdr['RA_APER']
+            dec_aper = ext1hdr['DEC_APER']
+            pa_aper = ext1hdr['PA_APER']
+        except KeyError:
+            cosutil.printWarning('One or more of RA_APER, DEC_APER, PA_APER is not present')
+            cosutil.printContinuation('in extension 1 header of file {}'.format(file))
+            return
+        diameter = 2.5 / 3600.0
+        s_region = 'CIRCLE {0:.8f} {1:.7f} {2:.8f}'.format(ra_aper, dec_aper, diameter)
```
Comment (nden) [inline costools/add_cos_s_region.py:74]: I believe the convention is also to include the coordinate frame btu you sould check with the archive what they expect. So something like 
```
'CIRCLE ICRS {0:.8f} {1:.7f} {2:.8f}'.format(ra_aper, dec_aper, diameter)
```

File: costools/add_cos_s_region.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    Filename ends with one of ['rawtag.fits',
+                               'rawtag_a.fits',
+                               'rawtag_b.fits',
+                               'rawaccum.fits',
+                               'rawaccum_a.fits',
+                               'rawaccum_b.fits',
+                               'rawacq.fits']
+
+    The S_REGION keyword will have the value
+    'CIRCLE ra_aper dec_aper diameter'
```
Comment (nden) [inline costools/add_cos_s_region.py:25]: See comment below about coordinate frame.

File: costools/add_cos_s_region.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+               'rawtag_b.fits',
+               'rawaccum.fits',
+               'rawaccum_a.fits',
+               'rawaccum_b.fits',
+               'rawacq.fits']
+
+    fitsfiles = glob.glob('*.fits')
+    for file in fitsfiles:
+        for ending in endings:
+            if file.endswith(ending):
```
Comment (nden) [inline costools/add_cos_s_region.py:45]: Is it necessary to iterate over `endings`? Is it sufficient to check if the specific file ending is in `endings`?

File: costools/add_cos_s_region.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+# Add the S_REGION keyword to raw files
+#
+import glob
+import argparse
+
+from calcos import cosutil
+from astropy.io import fits
+
+__version__ = '0.1'
```
[the remark this answers is not in the document]
Comment (stscirij) [reply costools/add_cos_s_region.py:11]: No, not necessary, I'll take it out

File: costools/add_cos_s_region.py (hunk 1 above)
Comment (stscirij) [reply costools/add_cos_s_region.py:74]: Will insert the ICRS.  STIS version already has it

File: costools/add_cos_s_region.py (hunk 3 above)
Comment (stscirij) [reply costools/add_cos_s_region.py:45]: This seemed like the easiest way.  It's too fiddly extracting the ending from a file name, as they can have different lengths and numbers of underscores

File: costools/add_cos_s_region.py (hunk 2 above)
Comment (stscirij) [reply costools/add_cos_s_region.py]: Fixed in latest version

**Review by stenhasselquist — APPROVED** [review]

**Pull request merged**